packages/sdk-middleware-auth/src/existing-token.js

```diff
@@ -0,0 +1,51 @@
+/* @flow */
+import type {
+  Middleware,
+  MiddlewareRequest,
+  MiddlewareResponse,
+  ExistingTokenMiddlewareOptions,
+  Next,
+} from '../../../types/sdk'
+
+export default function createAuthMiddlewareForExistingToken(
+  options: ExistingTokenMiddlewareOptions | string
+): Middleware {
+  return (next: Next) => (
+    request: MiddlewareRequest,
+    response: MiddlewareResponse
+  ) => {
+    const config = options
+      ? {
+          tokenType:
+            typeof options === 'object' && options !== null && options.tokenType
+              ? options.tokenType
+              : 'Bearer',
+          token: typeof options === 'string' ? options : options.token,
+          force: options.force !== undefined ? options.force : true,
+        }
+      : null
+
+    /** The request will not be modified if:
+     *  1. no argument is passed
+     *  2. an object is passed that doesn't contain a `token`
```

Comment:
Why would one not pass the `token` as part of the options. Shouldn't that be considered an error?

Reply:
In our CLI tools, accessToken is not mandatory. Enforcing it in this middleware will make it incompatible

Reply:
Isn't it then just a matter of only using this middleware when the user of the CLI provides their token?